amm/traits/stable_pool.rs

```diff
@@ -0,0 +1,204 @@
+use ink::prelude::vec::Vec;
+use ink::primitives::AccountId;
+use ink::LangError;
+use psp22::PSP22Error;
+
+use crate::MathError;
+
+#[ink::trait_definition]
+pub trait StablePoolView {
```

Comment:
I'm not very attached to names, we can change to "rated" everywhere. But after reading curves contracts, I realized that the notion of "rate" is already there, it's only that rates there are constant, and for us they are slowly changing. So calling it `StablePool` is fine imho. The name "rated pool" I find a little weird, because stable pools are also "rated", so it's not a distinguishing factor :)

Reply:
OK, that's fine.